account_account_constraint_code/models/account_account.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2022 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class AccountAccount(models.Model):
+
+    _inherit = "account.account"
+
+    @api.constrains("code")
+    def _disable_code_modification(self):
+        line = self.env["account.move.line"].search(
+            [("account_id", "in", self.ids)], limit=1
+        )
```

Comment:
Search with sudo? It's unlikely to be an issue here, but I always find it safer to check constraints with sudo so their effect does not depend on the permissions of the current user.